types_util.go

```diff
@@ -30,6 +30,8 @@ func walkNamedTypes(typ types.Type, callback func(*types.Named)) {
 		walkNamedTypes(typ.Elem(), callback)
 	case *types.Array:
 		walkNamedTypes(typ.Elem(), callback)
+	case *types.Alias:
+		walkNamedTypes(typ.Underlying(), callback)
```

Comment:
```suggestion
		walkNamedTypes(typ.Rhs(), callback)
```

From the docs:
> Rhs returns the type R on the right-hand side of an alias declaration "type A = R", which may be another alias.

[Underlying types](https://go.dev/ref/spec#Underlying_types) go too "deep".

For example, the underlying type of a `type Word = uint256.Int` is a `[4]uint64` instead of a `uint256.Int`.